src/vm/gcenv.ee.cpp

```diff
@@ -1429,3 +1429,94 @@ IGCToCLREventSink* GCToEEInterface::EventSink()
 
     return &g_gcToClrEventSink;
 }
+
+bool GCToEEInterface::CanEnableGCNumaAware()
+{
+    LIMITED_METHOD_CONTRACT;
+
+    return NumaNodeInfo::CanEnableGCNumaAware();
+}
+
+bool GCToEEInterface::GetNumaProcessorNodeEx(PPROCESSOR_NUMBER proc_no, uint16_t *node_no)
+{
+    LIMITED_METHOD_CONTRACT;
+
+    return NumaNodeInfo::GetNumaProcessorNodeEx(proc_no, node_no);
+}
+
+void* GCToEEInterface::VirtualAllocExNuma(void *lpAddr, size_t dwSize, uint32_t allocType, uint32_t prot, uint32_t node)
+{
+    LIMITED_METHOD_CONTRACT;
+
+    return NumaNodeInfo::VirtualAllocExNuma(GetCurrentProcess(), lpAddr, dwSize, allocType, prot, node);
```

Comment:
This goes to CoreCLR PAL. Once the local GC on Unix decides to free this memory, it will do it using `GCToOSInterface` that does not go to CoreCLR PAL. As it is written right now, this mismatch will cause bad things to happen. I think it would cause memory leak of CoreCLR PAL tracking structures in this case. 

This problem will get fixed by moving these method to `GCToOSInterface`.

Reply:
That's why I've suggested that this method should go to the GCToOSInterface too.